src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -400,5 +400,12 @@ private bool TryRegister(SafeSocketHandle socket, IntPtr handle, out Interop.Err
                 Interop.Sys.SocketEvents.Read | Interop.Sys.SocketEvents.Write, handle);
             return error == Interop.Error.SUCCESS;
         }
+
+        private class ThreadPoolResetEvent : ManualResetEventSlim, IThreadPoolWorkItem
+        {
+            public ThreadPoolResetEvent(bool initialState = false) : base(initialState) { }
```

Comment:
Seems like you could delete this ctor entirely, and just rely on the default one.